e2e-tests/src/test/button_game/mod.rs

```diff
@@ -55,6 +60,56 @@ pub fn marketplace(config: &Config) -> Result<()> {
     )?;
     marketplace.buy(&sign(&conn, player), None)?;
 
+    // Upgrade begins
```

Comment:
Request: I think this whole upgrade should be a separate test instead of embedded in the other test.